### PR TITLE
Add deprecations

### PR DESCRIPTION
**Pull request by ProfFan**
#927

### CODE REVIEW

File: gtsam/discrete/DiscreteConditional.cpp [hunk 1]
```diff
@@ -91,7 +91,7 @@ bool DiscreteConditional::equals(const DiscreteFactor& other,
     return false;
   else {
     const DecisionTreeFactor& f(
-        static_cast<const DecisionTreeFactor&>(other));
+        dynamic_cast<const DecisionTreeFactor&>(other));
```
Comment (varunagrawal) [inline gtsam/discrete/DiscreteConditional.cpp:94]: What's up with this? Was this needed?
Generally, static_cast is considered safer than dynamic_cast, hence asking.

File: gtsam/discrete/Potentials.cpp [hunk 2]
```diff
@@ -27,8 +27,8 @@ using namespace std;
 namespace gtsam {
 
 // explicit instantiation
-template class DecisionTree<Key, double>;
-template class AlgebraicDecisionTree<Key>;
+//template class GTSAM_EXPORT DecisionTree<Key, double>;
+//template class GTSAM_EXPORT AlgebraicDecisionTree<Key>;
```
Comment (varunagrawal) [inline gtsam/discrete/Potentials.cpp:31]: Accidentally commented out these two?

**Review by varunagrawal — CHANGES_REQUESTED** [review]
Couple of comments

File: gtsam/discrete/DiscreteConditional.cpp (hunk 1 above)
Comment (ProfFan) [reply gtsam/discrete/DiscreteConditional.cpp:94]: Oh yeah this is a clang-tidy false positive. I can revert this one

File: gtsam/discrete/Potentials.cpp (hunk 2 above)
Comment (ProfFan) [reply gtsam/discrete/Potentials.cpp:31]: I make the template GTSAM_EXPORT so these two are not needed. Need to check if Windows is happy with this or not...

Comment (varunagrawal) [reply gtsam/discrete/Potentials.cpp:31]: Gotcha. If they are not needed then kill them.

**Review by varunagrawal — APPROVED** [review]
LGTM

**Pull request merged**